nuxeo-runtime/nuxeo-runtime-reload/src/main/java/org/nuxeo/runtime/reload/ReloadComponent.java

```diff
@@ -157,53 +178,66 @@ public void flushSeamComponents() {
     }
 
     @Override
-    public String deployBundle(File file) throws BundleException {
-        return deployBundle(file, false);
-    }
-
-    @Override
-    public String deployBundle(File file, boolean reloadResourceClasspath) throws BundleException {
-        String name = getOSGIBundleName(file);
-        if (name == null) {
-            log.error(
-                    String.format("No Bundle-SymbolicName found in MANIFEST for jar at '%s'", file.getAbsolutePath()));
-            return null;
+    public void deployBundles(File[] files, boolean reloadResources) throws BundleException {
+        List<String> missingNames = Stream.of(files)
+                                          .filter(file -> getOSGIBundleName(file) == null)
+                                          .map(File::getAbsolutePath)
+                                          .collect(Collectors.toList());
+        if (!missingNames.isEmpty()) {
+            missingNames.forEach(
+                    name -> log.error(String.format("No Bundle-SymbolicName found in MANIFEST for jar at '%s'", name)));
+            // TODO investigate why we need to exit here, getBundleContext().installBundle(path) will throw an exception
+            // unless, maybe tests ?
+            return;
         }
-
-        String path = file.getAbsolutePath();
-
-        log.info(String.format("Before deploy bundle for file at '%s'\n" + "%s", path, getRuntimeStatus()));
-
-        if (reloadResourceClasspath) {
-            URL url;
-            try {
-                url = new File(path).toURI().toURL();
-            } catch (MalformedURLException e) {
-                throw new RuntimeException(e);
-            }
-            Framework.reloadResourceLoader(Collections.singletonList(url), null);
+        if (log.isInfoEnabled()) {
+            // TODO REVIEW - not logging each paths to avoid looping files once again, should we log still ?
+            StringBuilder builder = new StringBuilder("Before deploy bundles\n");
+            Framework.getRuntime().getStatusMessage(builder);
+            log.info(builder.toString());
```

Comment:
I will keep the status at INFO level as we want to print the errors and warnings happened during hot reload.
Not sure if it's interesting to log at DEBUG level the list of files, I will just drop the TODO.